pax-chassis-web/interface/src/classes/native-element-pool.ts

```diff
@@ -346,14 +365,32 @@ export class NativeElementPool {
     textUpdate(patch: TextUpdatePatch) {
         let leaf = this.nodesLookup.get(patch.id!) as HTMLElement;
         let textChild = leaf!.firstChild as HTMLElement;
+        // should be start listening to this elements size and
+        // send interrupts to the engine, or not?
+        let start_listening = false;
+
         // Handle size_x and size_y
         if (patch.size_x != null) {
-            leaf!.style.width = patch.size_x + "px";
+
+            // if size_x = -1.0, the engine wants to know
+            // this elements size from the chassi.
+            if (patch.size_x == -1.0) {
```

Comment:
Let's not use a magic number ("-1"); instead let's send this information descriptively in the native message. Use an enum that tells the what sort of bounding box this text node uses (fixed or auto-resize).

Reply:
I've been keeping AutoWith and AutoHeight Separate, do you prefer if they are one thing? Originally I wanted to send Option<number>, but the fields themselves are already optional and wasn't sure how to get double optionals across the wasm bridge, I can look in to that option as well. Two enums or two true/false values is another option, but that's more info than needed (one could accidentally send AutoSized + a width, which is "illegal).

Reply:
I see. Yeah it might be difficult to do an option of an option. Okay let's leave it as is for now.